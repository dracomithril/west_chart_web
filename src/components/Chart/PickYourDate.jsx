/**
 * Created by Gryzli on 12.02.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { OverlayTrigger, Tooltip, Button } from 'react-bootstrap';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

const DateButton = props => <Button onClick={props.onClick}>{props.value}</Button>;
DateButton.propTypes = {
  onClick: PropTypes.func,
  value: PropTypes.string,
};

const PickYourDate = ({ start_date, show_last, onDaysChange, onDateChange }) => (
  <div id="pickYourDate">
    <OverlayTrigger
      trigger={['hover', 'focus']}
      overlay={<Tooltip id="go_back">How far in time you will travel</Tooltip>}
    >
      <span style={{ paddingRight: 10, paddingBottom: 5 }}>
        {'go back '}
        <input
          className="num_days"
          type="number"
          name="show_last"
          min={0}
          max={62}
          value={show_last}
          step={1}
          onChange={({ target }) => onDaysChange(target)}
        />
        {' days'}
      </span>
    </OverlayTrigger>
    <br />
    <div className="datePicker">
      starting from
      <DatePicker
        selected={start_date}
        withPortal
        dateFormat="DD/MM/YY"
        onChange={onDateChange}
        customInput={<DateButton />}
      />
    </div>
  </div>
);
PickYourDate.propTypes = {
  start_date: PropTypes.object,
  show_last: PropTypes.number,
  onDaysChange: PropTypes.func,
  onDateChange: PropTypes.func,
};
export default PickYourDate;
