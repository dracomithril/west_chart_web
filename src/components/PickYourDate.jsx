/**
 * Created by Gryzli on 12.02.2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Button, Checkbox, OverlayTrigger, Popover, Tooltip } from 'react-bootstrap';
import DatePicker from 'react-datepicker';
import action_types from './../reducers/action_types';

export default class PickYourDate extends React.Component {
  /* istanbul ignore next */
  componentDidMount() {
    console.info('component ChartPresenter did mount');
  }

  /* istanbul ignore next */
  componentWillUnmount() {
    console.info('component ChartPresenter unmounted');
  }

  render() {
    const { store } = this.context;
    const { enable_until, start_date, show_last } = store.getState();
    const footer = (
      <div className="datePiker">
        <Checkbox
          checked={enable_until}
          name="enable_until"
          onChange={({ target }) =>
            store.dispatch({
              type: action_types.TOGGLE_ENABLE_UNTIL,
              checked: target.checked,
            })
          }
        >
          {'Use date: '}
        </Checkbox>
        <DatePicker
          selected={start_date}
          dateFormat="DD/MM/YYYY"
          onChange={date => {
            store.dispatch({ type: action_types.UPDATE_START_TIME, date });
          }}
          disabled={!enable_until}
        />
      </div>
    );
    return (
      <div id="pickYourDate">
        <OverlayTrigger
          trigger={['hover', 'focus']}
          overlay={<Tooltip id="go_back">How far in time you will travel</Tooltip>}
        >
          <span style={{ paddingRight: 10 }}>
            {'go back '}
            <input
              className="num_days"
              type="number"
              name="show_last"
              min={0}
              max={62}
              value={show_last}
              step={1}
              onChange={({ target }) => {
                store.dispatch({ type: action_types.UPDATE_SHOW_LAST, days: Number(target.value) });
              }}
            />
            {' days'}
          </span>
        </OverlayTrigger>
        <OverlayTrigger
          trigger="click"
          placement="bottom"
          overlay={<Popover id="more_options_chose_date">{footer}</Popover>}
        >
          <Button className="fa fa-angle-double-down" bsSize="small" />
        </OverlayTrigger>
      </div>
    );
  }
}
PickYourDate.contextTypes = {
  store: PropTypes.object.isRequired,
};
PickYourDate.propTypes = {};
