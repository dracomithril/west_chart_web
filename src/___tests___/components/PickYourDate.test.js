/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from "react";
import PropTypes from "prop-types";
import PickYourDate from "./../../components/PickYourDate";
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from "enzyme-to-json";
const utils = require('../../utils/utils');

import configureMockStore from "redux-mock-store";
Enzyme.configure({ adapter: new Adapter() });
jest.mock('./../../utils');
const initial_state = require('../data/initial_state.json');
const mockStore = configureMockStore([]);
describe('<PickYourDate/>', () => {
  afterAll(() => {
    jest.unmock('./../../utils');
  });
  it('renders without crashing', () => {
    const store = mockStore(initial_state);
    const wrapper = shallow(
      <PickYourDate/>, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  xit('is able to click btn', () => {
    const store = mockStore(initial_state);
    const wrapper = shallow(
      <PickYourDate/>, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    let button = wrapper.find('#updateChartB');

    button.simulate('click');
    let actions = store.getActions();
    expect(actions.length).toBe(0);
    expect(utils.UpdateChart.mock.calls.length).toBe(1);
  });
});
