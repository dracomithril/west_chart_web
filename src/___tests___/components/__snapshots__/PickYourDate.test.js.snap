// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PickYourDate/> renders without crashing 1`] = `
<div
  id="pickYourDate"
>
  <OverlayTrigger
    defaultOverlayShown={false}
    overlay={
      <Tooltip
        bsClass="tooltip"
        id="go_back"
        placement="right"
      >
        How far in time you will travel
      </Tooltip>
    }
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <span
      style={
        Object {
          "paddingRight": 10,
        }
      }
    >
      go back 
      <input
        className="num_days"
        max={62}
        min={0}
        name="show_last"
        onChange={[Function]}
        step={1}
        type="number"
        value={31}
      />
       days
    </span>
  </OverlayTrigger>
  <OverlayTrigger
    defaultOverlayShown={false}
    overlay={
      <Popover
        bsClass="popover"
        id="more_options_chose_date"
        placement="right"
      >
        <div
          className="datePiker"
        >
          <Checkbox
            bsClass="checkbox"
            checked={undefined}
            disabled={false}
            inline={false}
            name="enable_until"
            onChange={[Function]}
            title=""
          >
            Use date: 
          </Checkbox>
          <DatePicker
            allowSameDay={false}
            dateFormat="DD/MM/YYYY"
            dateFormatCalendar="MMMM YYYY"
            disabled={true}
            disabledKeyboardNavigation={false}
            dropdownMode="scroll"
            monthsShown={1}
            onBlur={[Function]}
            onChange={[Function]}
            onClickOutside={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMonthChange={[Function]}
            onSelect={[Function]}
            onYearChange={[Function]}
            preventOpenOnFocus={false}
            selected={undefined}
            shouldCloseOnSelect={true}
            showTimeSelect={false}
            timeIntervals={30}
            withPortal={false}
          />
        </div>
      </Popover>
    }
    placement="bottom"
    trigger="click"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsSize="small"
      bsStyle="default"
      className="fa fa-angle-double-down"
      disabled={false}
    />
  </OverlayTrigger>
</div>
`;
